src/plots/cartesian/axes.js

```diff
@@ -1075,6 +1080,49 @@ axes.calcTicks = function calcTicks(ax, opts) {
         }
     }
 
+    // check if ticklabelIndex makes sense, otherwise ignore it
+    if(!minorTickVals || minorTickVals.length < 2) {
+        ticklabelIndex = false;
+    } else {
+        var diff = minorTickVals[1].value - minorTickVals[0].value;
```

Comment:
Should we consider the `reversed` axes here?
Right now on `zzz_ticklabelindex` mock if I reverse the axis direction:
```js
Plotly.relayout(gd, 'xaxis2.autorange', 'reversed');
```
I get labels on January not December!

Reply:
Thanks for spotting this. I added the check and added a test case in a mock.